From 634428523a1c72ebab27d3a5797da5aa21700b80 Mon Sep 17 00:00:00 2001
From: Wu Guoxing <b02248@freescale.com>
Date: Thu, 26 Jun 2008 13:23:58 +0800
Subject: [PATCH] ENGR00077820 The image is covered by color stripes

V4L2:The image is covered by color stripes when CMOS captures.

Signed-off-by: Guoxing Wu <b02248@freescale.com>
---
 arch/arm/mach-mx35/clock.c       |    2 +-
 arch/arm/mach-mx35/mx35_3stack.c |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-mx35/clock.c b/arch/arm/mach-mx35/clock.c
index 6be1d36..c304883 100644
--- a/arch/arm/mach-mx35/clock.c
+++ b/arch/arm/mach-mx35/clock.c
@@ -1387,7 +1387,7 @@ static struct clk wdog_clk = {
 
 static struct clk csi_clk = {
 	.name = "csi_clk",
-	.parent = &peri_pll_clk,
+	.parent = &cpu_clk,
 	.recalc = _clk_csi_recalc,
 	.round_rate = _clk_round_rate,
 	.set_rate = _clk_csi_set_rate,
diff --git a/arch/arm/mach-mx35/mx35_3stack.c b/arch/arm/mach-mx35/mx35_3stack.c
index ef5ef35..ae98f26 100644
--- a/arch/arm/mach-mx35/mx35_3stack.c
+++ b/arch/arm/mach-mx35/mx35_3stack.c
@@ -267,7 +267,7 @@ static struct mxc_camera_platform_data camera_data = {
 	.io_regulator = NULL,
 	.analog_regulator = "LDO7",
 	.gpo_regulator = NULL,
-	.mclk = 24000000,
+	.mclk = 20000000,
 };
 
 void si4702_reset(void)
-- 
1.5.4.4

